code/modules/research/ordnance/doppler_array.dm

```diff
@@ -0,0 +1,305 @@
+	var/datum/data/tachyon_record/new_record = new /datum/data/tachyon_record()
+	new_record.name = "Log Recording #[record_number]"
+	new_record.timestamp = station_time_timestamp()
+	new_record.coordinates = "[epicenter.x], [epicenter.y]"
+	new_record.displacement = took
+	new_record.factual_radius["epicenter_radius"] = devastation_range
+	new_record.factual_radius["outer_radius"] = heavy_impact_range
+	new_record.factual_radius["shockwave_radius"] = light_impact_range
+	new_record.reaction_results = fetched_reaction_results
+	new_record.explosion_identifier = explosion_index
+
+	var/list/messages = list("Explosive disturbance detected.",
+		"Epicenter at: grid ([epicenter.x], [epicenter.y]). Temporal displacement of tachyons: [took] seconds.",
+		"Factual: Epicenter radius: [devastation_range]. Outer radius: [heavy_impact_range]. Shockwave radius: [light_impact_range].",
+	)
+
+	// If the bomb was capped, say its theoretical size.
+	if(devastation_range < orig_dev_range || heavy_impact_range < orig_heavy_range || light_impact_range < orig_light_range)
+		messages += "Theoretical: Epicenter radius: [orig_dev_range]. Outer radius: [orig_heavy_range]. Shockwave radius: [orig_light_range]."
+		new_record.theory_radius["epicenter_radius"] = orig_dev_range
+		new_record.theory_radius["outer_radius"] = orig_heavy_range
+		new_record.theory_radius["shockwave_radius"] = orig_light_range
+
+	for(var/message in messages)
+		say(message)
+
+	record_number++
+	records += new_record
+
+	return TRUE
+
+/obj/machinery/doppler_array/proc/eject_disk(mob/user)
+	if(!inserted_disk)
+		return FALSE
+	if(user)
+		user.put_in_hands(inserted_disk)
+	else
+		inserted_disk.forceMove(drop_location())
+	playsound(src, 'sound/machines/card_slide.ogg', 50)
+	return TRUE
+
+/// We rely on exited to clear references.
+/obj/machinery/doppler_array/Exited(atom/movable/gone, direction)
+	if(gone == inserted_disk)
+		inserted_disk = null
+	. = ..()
+
+/obj/machinery/doppler_array/powered()
+	return (!panel_open && ..())
+
+/obj/machinery/doppler_array/update_overlays()
+	. = ..()
+	if(panel_open)
+		. += mutable_appearance(icon, "[base_icon_state]_open")
+	else
+		. += mutable_appearance(icon,"[base_icon_state]_cable")
+
+	if(machine_stat & BROKEN) // Probably meant to be used on an indestructible doppler, but this is not implemented.
+		. += mutable_appearance(icon, "[base_icon_state]_screen-broken")
+	else if (machine_stat & NOPOWER)
+		. += mutable_appearance(icon, "[base_icon_state]_screen-off")
+
+/obj/machinery/doppler_array/on_deconstruction()
+	eject_disk()
+	. = ..()
+
+/obj/machinery/doppler_array/Destroy()
+	inserted_disk = null
+	QDEL_NULL(records) //We only want the list nuked, not the contents.
+	. = ..()
+
+/obj/machinery/doppler_array/proc/update_doppler_light()
+	SIGNAL_HANDLER
+	set_light_on(!(machine_stat & NOPOWER))
+
+/obj/machinery/doppler_array/AltClick(mob/user)
+	return ..() // This hotkey is BLACKLISTED since it's used by /datum/component/simple_rotation
```

Comment:
What?

Reply:
https://github.com/tgstation/tgstation/blob/da5d271e7313fe1e1169723cf9abb62665dbf9b6/code/game/machinery/doppler_array.dm#L128-L129

Reply:
but... why?

Reply:
Figured its a cool warning. Want me to kill it?

Reply:
I just don't really understand it. seems like a strange place to put it

Reply:
so, verdict? want it dead?

Reply:
you can leave it

Reply:
Alt click rotation is still there though.

Reply:
Oh I see. Gahhh
Yeah just leave it for now